warc_metadata_sidecar.py

```diff
@@ -0,0 +1,207 @@
+                    record.rec_headers, rawPayload,
+                    record.http_headers, record.content_type,
+                    record.length
+                    )
+                decodedPayload = recordCopy.content_stream().read()
+                rawPayload.seek(0)
+            else:
+                decodedPayload = rawPayload
+
+            print(url)
+            find_mime_and_puid(fido, rawPayload, url)
+            result_dict = {'encoding': None}
+
+            if fido.mime:
+                image_mimes = re.compile(r'(jpeg|gif|png)')
+                text_format_mimes = re.compile(r'(text|html|xml)')
+
+                if not image_mimes.search(fido.mime):
+                    result_dict = find_character_set(decodedPayload)
+
+                if text_format_mimes.search(fido.mime):
+                    lang_cld = find_language(decodedPayload)
+                    text_mime += 1
+                    string_payload = '{0} {1}\n{2} {3}\n{4} {5}\n{6} {7}'.format(
+                        mime_title, fido.mime,
+                        puid_title, fido.puid,
+                        charset_title, result_dict,
+                        language_title, lang_cld
+                        )
+                elif result_dict['encoding'] is None:
+                    non_text += 1
+                    string_payload = '{0} {1}\n{2} {3}'.format(
+                        mime_title, fido.mime,
+                        puid_title, fido.puid,
+                        )
+                else:
+                    non_text += 1
+                    string_payload = '{0} {1}\n{2} {3}\n{4} {5}'.format(
+                        mime_title, fido.mime,
+                        puid_title, fido.puid,
+                        charset_title, result_dict,
+                        )
+            elif result_dict['encoding'] is None:
+                non_mime += 1
+                string_payload = ''
+            else:
+                non_mime += 1
+                string_payload = '{0} {1}'.format(charset_title, result_dict)
+
+            byte_payload = bytearray(string_payload.encode())
+            writer = WARCWriter(output, gzip=False)  # gzip will equal True
+            meta_record = writer.create_warc_record(url,
+                                                    'metadata',
+                                                    payload=io.BytesIO(byte_payload),
+                                                    warc_headers_dict=warc_dict
+                                                    )
+            writer.write_record(meta_record)
+
+        logging.info('Found %s record(s)', record_count)
+    logging.info('Finished creating sidecar in %s', str(timedelta(seconds=(time.time() - start))))
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'warc_file',
+        action='store',
+        help='a warc file that will be used to generate a sidecar with metadata'
```

Comment:
Anytime in this file in comments or strings where we refer to a warc file, I suggest capitalizing "WARC".